examples/asr/librispeech_emformer_rnnt/global_stats.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+import pathlib
+from argparse import ArgumentParser
+
+import torch
+import torchaudio
+from utils import GAIN, piecewise_linear_log, spectrogram_transform
+
+logger = logging.getLogger()
+
+
+def generate_statistics(samples):
+    E_x = 0
+    E_x_2 = 0
+    N = 0
+
+    for idx, sample in enumerate(samples):
+        mel_spec = spectrogram_transform(sample[0].squeeze()).transpose(1, 0)
+        scaled_mel_spec = piecewise_linear_log(mel_spec * GAIN)
+        sum = scaled_mel_spec.sum(0)
+        sq_sum = scaled_mel_spec.pow(2).sum(0)
+        M = scaled_mel_spec.size(0)
+
+        E_x = E_x * (N / (N + M)) + sum / (N + M)
+        E_x_2 = E_x_2 * (N / (N + M)) + sq_sum / (N + M)
+        N += M
+
+        if idx % 100 == 0:
+            logger.info(f"Processed {idx}")
+
+    return E_x, (E_x_2 - E_x ** 2) ** 0.5
+
+
+def cli_main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        "--librispeech_path",
+        required=True,
+        type=pathlib.Path,
+        help="Path to LibriSpeech datasets. "
+        "All of 'train-clean-360', 'train-clean-100', and 'train-other-500' must exist.",
+    )
+    parser.add_argument(
+        "--output_path",
+        default=pathlib.Path("global_stats.json"),
+        type=pathlib.Path,
+        help="File to save feature statistics to. (Default: './global_stats.json')",
+    )
+    args = parser.parse_args()
```

Comment:
nit: I would extract the logic to parse CLI arguments to a helper function. That way the execution flow of the main function becomes easier to follow.